src/plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -827,26 +823,27 @@ def _compute_cross_correlation(self, coeffs_tensor: List[Tensor],
             # Then normalize it to get the Pearson product-moment correlation
             # coefficient, see
             # https://numpy.org/doc/stable/reference/generated/numpy.corrcoef.html.
-            # First, compute the variances of each coeff (if coeff and
-            # coeff_other are identical, this is equivalent to the diagonal of
-            # the above covar matrix, but re-computing it is actually faster)
-            coeff_var = einops.einsum(coeff, coeff,
-                                      'b c o1 h w, b c o1 h w -> b c o1')
-            coeff_var = coeff_var / numel
-            coeffs_var.append(coeff_var)
-            if tensors_are_identical:
-                coeff_other_var = coeff_var
+            if coeffs_var is None:
+                # First, compute the variances of each coeff
+                coeff_var = einops.einsum(coeff, coeff,
```

Comment:
Good eye. Yes, they are. `coeff` is either the demeaned magnitude of the steerable pyramid coefficients or the real component of those coefficients (both computed in `_compute_intermediate_representations`). The magnitude is explicitly mean-centered, like I said, and the steerable pyramid coefficients all have mean=0, because they're the output of a bandpass filter.

This is not generally true, as you point out, hence why this is a method of this class, instead of a more general function.